BraintreeCore/src/main/java/com/braintreepayments/api/ErrorWithResponse.kt

```diff
@@ -0,0 +1,157 @@
+package com.braintreepayments.api
+
+import android.os.Parcel
+import android.os.Parcelable
+import androidx.annotation.RestrictTo
+import com.braintreepayments.api.GraphQLConstants.ErrorMessages
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Error container returned when the Braintree server returns a 422 Unprocessible Entity.
+ * A 422 occurs when a request is properly formed, but the server was unable to take the requested
+ * action due to bad user data.
+ *
+ * ErrorWithResponse parses the server's error response and exposes the errors.
+ *
+ * @property statusCode HTTP status code from the Braintree gateway.
+ * @property message Human readable top level summary of the error.
+ * @property errorResponse The full error response as a [String].
+ * @property fieldErrors All the field errors.
+ */
+open class ErrorWithResponse : Exception, Parcelable {
+
+    open var statusCode: Int = 0
+        internal set
+
+    private var _message: String? = null
+    override val message: String?
+        get() = _message
```

Comment:
Do we really need this backing property for `message`?

Reply:
Technically yes, there's some weirdness with the Java interop here. The `message` property is an override from the `Exception` class and the `super` constructor is never called in the current implementation. I expect we'll refactor this class sometime after the Kotlin migration.